packages/mui-base/src/unstable_useNumberInput/useNumberInput.ts

```diff
@@ -400,12 +403,42 @@ export function useNumberInput(parameters: UseNumberInputParameters): UseNumberI
     };
   };
 
-  const handleStepperButtonMouseDown = (event: React.PointerEvent) => {
-    event.preventDefault();
+  const recurse = (payload: {
+    actionType: typeof NumberInputActionTypes.increment | typeof NumberInputActionTypes.decrement;
+    event: React.PointerEvent;
+  }) => {
+    clearTimeoutRef();
 
-    if (inputRef.current) {
-      inputRef.current.focus();
-    }
+    const { actionType, event } = payload;
+
+    timeoutRef.current = window.setTimeout(() => {
```

Comment:
Should clear this timer on unmount?

Reply:
I found I could use `useTimeout` from https://github.com/mui/material-ui/pull/39060 which has all the optimizations built in ~